### PR TITLE
Initial setup of Track 2 containerregistry code in repo

### PR DESCRIPTION
**Pull request by annelo-msft**
This is the initial push of code to sdk/containerregistry/... for the new Track 2 library.

This includes ci.yml and tests.yml to set up the CI pipeline.

### CODE REVIEW

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Azure.Containers.ContainerRegistry.csproj [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <TargetFrameworks>$(RequiredTargetFrameworks)</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="System.Text.Json" />
+  </ItemGroup>
+
+  <!-- Shared source from Azure.Core -->
+  <ItemGroup>
+    <Compile Include="$(AzureCoreSharedSources)ArrayBufferWriter.cs" Link="Shared\%(RecursiveDir)\%(Filename)%(Extension)" />
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Azure.Containers.ContainerRegistry.csproj:16]: nit: This is the syntax I've seen used most recently.

```suggestion
        <Compile Include="$(AzureCoreSharedSources)ArrayBufferWriter.cs" LinkBase="Shared\Core" />(Extension)" />
```

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/autorest.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Generated code configuration
+
+Run `dotnet build /t:GenerateCode` to generate code.
+
+``` yaml
+input-file:
```
Comment (christothes) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/autorest.md:6]: I think this should be linking to https://github.com/Azure/azure-rest-api-specs/blob/master/specification/containerregistry/data-plane/readme.md

see the following example:
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/tables/Azure.Data.Tables/src/autorest.md

**Review by christothes — APPROVED** [review]

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/autorest.md (hunk 2 above)
Comment (annelo-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/autorest.md:6]: Thanks, @christothes !  I'll work with @ShivangiReja on this.

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/Directory.Build.props [hunk 3]
```diff
@@ -0,0 +1,3 @@
+﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory).., Directory.Build.props))\Directory.Build.props" />
```
Comment (maririos) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/Directory.Build.props:2]: do you know why this is needed? I see it in the template and in FR too.
I went to FR and deleted the file and was able to build and run all tests.. so idk

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Azure.Containers.ContainerRegistry.csproj (hunk 1 above)
Comment (maririos) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Azure.Containers.ContainerRegistry.csproj]: @christothes  what is the difference between `Link` and `LinkBase`?

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+
+[assembly: Azure.Core.AzureResourceProviderNamespace("Microsoft.ContainerRegistry")]
```
Comment (maririos) [inline sdk/containerregistry/Azure.Containers.ContainerRegistry/src/properties/AssemblyInfo.cs:6]: @annelo-msft  could you remind me why is this needed?

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Azure.Containers.ContainerRegistry.csproj (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Azure.Containers.ContainerRegistry.csproj]: LinkBase allows you to skip the filename part of the path.

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/Directory.Build.props (hunk 3 above)
Comment (JoshLove-msft) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/Directory.Build.props:2]: I think it probably isn't needed if you don't need to add anything specific to it for your library, but the template includes it in case you do.

File: sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Azure.Containers.ContainerRegistry.csproj (hunk 1 above)
Comment (maririos) [reply sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Azure.Containers.ContainerRegistry.csproj]: thank you!
